src/ray/util/logging.h

```diff
@@ -0,0 +1,145 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#ifndef RAY_UTIL_LOGGING_H
+#define RAY_UTIL_LOGGING_H
+
+#include <cstdlib>
+#include <iostream>
+
+#include "ray/util/macros.h"
+
+namespace ray {
+
+// Stubbed versions of macros defined in glog/logging.h, intended for
+// environments where glog headers aren't available.
+//
+// Add more as needed.
+
+// Log levels. LOG ignores them, so their values are abitrary.
+
+#define RAY_DEBUG (-1)
+#define RAY_INFO 0
+#define RAY_WARNING 1
+#define RAY_ERROR 2
+#define RAY_FATAL 3
+
+#define RAY_LOG_INTERNAL(level) ::ray::internal::CerrLog(level)
+#define RAY_LOG(level) RAY_LOG_INTERNAL(RAY_##level)
+#define RAY_IGNORE_EXPR(expr) ((void)(expr));
+
+#define RAY_CHECK(condition)                           \
+  (condition) ? 0                                        \
+              : ::ray::internal::FatalLog(RAY_FATAL) \
+                    << __FILE__ << __LINE__ << " Check failed: " #condition " "
```

Comment:
Does this still give a backtrace?

Reply:
I added a TODO for this

Reply:
We can just use the current `CHECK` macro, right? This functionality seems too important to lose.

Reply:
Another thing our current macro does is that it calls `abort()` in order to generate core dumps. This is pretty important. What's the advantage of the new macro?

Reply:
I reintroduced the backtrace now :)